src/main/java/com/minecolonies/coremod/colony/buildings/registry/BuildingRegistry.java

```diff
@@ -0,0 +1,303 @@
+package com.minecolonies.coremod.colony.buildings.registry;
+
+import com.minecolonies.api.util.BlockPosUtil;
+import com.minecolonies.api.util.Log;
+import com.minecolonies.coremod.blocks.*;
+import com.minecolonies.coremod.colony.Colony;
+import com.minecolonies.coremod.colony.ColonyView;
+import com.minecolonies.coremod.colony.buildings.*;
+import com.minecolonies.coremod.colony.buildings.views.AbstractBuildingView;
+import com.minecolonies.coremod.colony.buildings.workerbuildings.*;
+import com.minecolonies.coremod.colony.workorders.WorkOrderBuildBuilding;
+import com.minecolonies.coremod.entity.ai.citizen.builder.ConstructionTapeHelper;
+import com.minecolonies.coremod.tileentities.TileEntityColonyBuilding;
+import com.minecolonies.coremod.util.ColonyUtils;
+import com.minecolonies.coremod.util.StructureWrapper;
+import io.netty.buffer.ByteBuf;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.util.Tuple;
+import net.minecraft.util.math.BlockPos;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.TreeMap;
+
+import static com.minecolonies.api.util.constant.NbtTagConstants.TAG_BUILDING_TYPE;
+import static com.minecolonies.api.util.constant.NbtTagConstants.TAG_LOCATION;
+
+/**
+ * Class taking care of registering the buildings.
+ */
+public class BuildingRegistry
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Make this class "final" or add a public constructor. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2974)